pytools/convergence.py

```diff
@@ -20,13 +22,32 @@ def estimate_order_of_convergence(abscissae, errors):
 
 
 class EOCRecorder:
+    """
+    .. automethod:: add_data_point
+
+    .. automethod:: estimate_order_of_convergence
+    .. automethod:: order_estimate
+    .. automethod:: max_error
+
+    .. automethod:: pretty_print
+    .. automethod:: write_gnuplot_file
+    """
+
     def __init__(self):
-        self.history = []
+        self.history: List[Tuple[float, float]] = []
+
+    def add_data_point(self, abscissa: float, error: float) -> None:
+        from numbers import Number
+        assert isinstance(abscissa, Number), \
+                f"'abscissa' is not a number: {type(abscissa).__name__}"
+        assert isinstance(error, Number), \
+                f"'error' is not a number: {type(error).__name__}"
```

Comment:
Input checking should be `if` not `assert`.

Reply:
Done in new version 3bd3f56.